pipelines/scripts/synapse_artifact/synapse_database_util.py

```diff
@@ -0,0 +1,115 @@
+from typing import List, Dict, Any
+import json
+
+class SynapseLakeDatabaseUtil(SynapseArtifactUtil):
+    """
+    Class for managing the retrieval and analysis of Synapse Lake Database artifacts
+    """
+
+    @classmethod
+    def get_type_name(cls) -> str:
+        return "database"
+
+    def get(self, artifact_name: str, **kwargs: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Retrieves a specific Lake Database definition by name.
+        """
+        return self._web_request(
+            f"{self.synapse_endpoint}/databases?/{artifact_name}?api-version=2021-04-01",
+        ).json()
+
+    def get_all(self,**kwargs: Dict[str, Any])-> List[Dict[str, Any]]:
+        """
+        Retrieves all Lake Database definitions in the workspace.
+        Handles empty or invalid responses gracefully.
+        """
+        url = f"{self.synapse_endpoint}/databases?api-version=2021-04-01"
+        response = self._web_request(url)
+
+        # If the response has no content (204) or an empty body, return empty list
+        if not response.text.strip():
+            return []
+
+        try:
+            data = response.json()
+        except ValueError:
+            raise RuntimeError(f"Invalid JSON returned from {url}: {response.text[:200]}")
+
+        all_lake_dbs = data["items"]
+
+        while "nextLink" in data:
+            next_link = data["nextLink"]
+            next_response = self._web_request(next_link)
+
+            if not next_response.text.strip():
+                break
+
+            try:
+                data = next_response.json()
+            except ValueError:
+                raise RuntimeError(f"Invalid JSON returned from {next_link}: {next_response.text[:200]}")
+
+            all_lake_dbs.extend(data.get("value", []))
+
+        return all_lake_dbs
+    
+    def get_tables(self, database_name: str) -> List[Dict[str, Any]]:
+        url = f"{self.synapse_endpoint}/databases/{database_name}/tables?api-version=2021-04-01"
+        response = self._web_request(url)
+
+        if not response.text.strip():
+            return []
+
+        try:
+            data = response.json()
+        except ValueError:
+            raise RuntimeError(f"Invalid JSON from {url}: {response.text[:200]}")
+
+        all_tables = data.get("items", []) or data.get("value", [])
+
+        while "nextLink" in data:
+            next_link = data["nextLink"]
+            next_response = self._web_request(next_link)
+            if not next_response.text.strip():
+                break
+            data = next_response.json()
+            all_tables.extend(data.get("items", []) or data.get("value", []))
+
+        return all_tables
+    
+    def get_table(self, database_name: str, table_name: str) -> Dict[str, Any]:
+        url = f"{self.synapse_endpoint}/databases/{database_name}/tables/{table_name}?api-version=2021-04-01"
+        response = self._web_request(url)
+        return response.json()
```

Comment:
These probably need to be moved to a new util class, since tables seem to be a different artifact kind